networkdb/delegate.go

```diff
@@ -163,11 +163,27 @@ func (nDB *NetworkDB) handleNetworkEvent(nEvent *NetworkEvent) bool {
 	}
 
 	if n, ok := nodeNetworks[nEvent.NetworkID]; ok {
+		// If a leave event is received late network state shouldn't
+		// be updated, but cleanup the old table entries.
+		if nEvent.Type == NetworkEventTypeLeave {
+			flushEntries = true
+		}
 		// We have the latest state. Ignore the event
 		// since it is stale.
 		if n.ltime >= nEvent.LTime {
 			return false
 		}
+		// On every node epoch ID is a monotonously increasing
+		// number. Its safe to discard older events if we have
+		// a higher number already.
+		if nEvent.EpochID != 0 && nEvent.EpochID <= n.epochID {
+			return false
+		}
+		// On a network join from a node, cleanup its table entries
+		// for this network with Epoch ID < current Epoch ID.
+		// On a network leave from a node, cleanup its table entries
+		// with Epoch ID <= current Epoch ID
+		flushEntries = true
```

Comment:
if we are doing the flush all the time, do we really need a boolean, can we simply remove it?
Also with the current code that witness all the messages, this operation will be done every bulk sync. Till we don't fix that, do you think this can create a problem, stalling the whole list of entries for seconds while it is processing them?